gazebo_plugins/src/gazebo_ros_wheel_slip.cpp

```diff
@@ -114,6 +132,103 @@ void GazeboRosWheelSlip::Load(gazebo::physics::ModelPtr _model, sdf::ElementPtr
   impl_->ros_node_->declare_parameter(
     "slip_compliance_unitless_longitudinal",
     slip_longitudinal_default);
+
+  double zero_wheel_spin_tolerance = 1.0e-3;
+  if (!_sdf->HasElement("zero_wheel_spin_tolerance")) {
+    RCLCPP_INFO(
+      impl_->ros_node_->get_logger(), "Missing <zero_wheel_spin_tolerance>, defaults to %f",
+      zero_wheel_spin_tolerance);
+  } else {
+    zero_wheel_spin_tolerance = _sdf->GetElement("zero_wheel_spin_tolerance")->Get<double>();
+  }
+
+  double update_rate = 100.0;
+  if (!_sdf->HasElement("update_rate")) {
+    RCLCPP_INFO(
+      impl_->ros_node_->get_logger(), "Missing <update_rate>, defaults to %f", update_rate);
+  } else {
+    update_rate = _sdf->GetElement("update_rate")->Get<double>();
+  }
+
+  if (update_rate > 0.0) {
+    impl_->update_period_ = 1.0 / update_rate;
+  } else {
+    impl_->update_period_ = 0.0;
+  }
+
+  impl_->last_update_time_ = _model->GetWorld()->SimTime();
+
+  impl_->slip_publisher_ = impl_->ros_node_->create_publisher<gazebo_msgs::msg::InstantSlip>(
+    "wheelslip", 10);
+
+  auto on_update_callback = [this, zero_wheel_spin_tolerance](
+    const gazebo::common::UpdateInfo & info) {
+
+    #ifdef IGN_PROFILER_ENABLE
+      IGN_PROFILE("instant slip callback");
+    #endif
+
+    gazebo::common::Time current_time = info.simTime;
+
+    // If the world is reset, for example
+    if (current_time < impl_->last_update_time_) {
+      RCLCPP_INFO(impl_->ros_node_->get_logger(), "Negative sim time difference detected.");
+      impl_->last_update_time_ = current_time;
+    }
+
+    // Check period
+    double seconds_since_last_update = (current_time - impl_->last_update_time_).Double();
+
+    if (seconds_since_last_update < impl_->update_period_) {
+      return;
+    }
+
+    #ifdef IGN_PROFILER_ENABLE
+      IGN_PROFILE_BEGIN("fill ROS message");
+    #endif
+
+    // Populate message
+    auto slip_msg = gazebo_msgs::msg::InstantSlip();
+
+    std::map<std::string, ignition::math::Vector3d> slips;
+    this->GetSlips(slips);
+    for(const auto& wheel : slips)
+    {
+      auto name = wheel.first;
+      auto slip = wheel.second;
+      auto spin_speed = slip.Z();
+      auto long_vel = slip.X() + slip.Z();
+      auto lat_vel = slip.Y();
+      double long_slip;
+      if (abs(spin_speed) < zero_wheel_spin_tolerance) {
```

Comment:
I think you want `fabs` instead of `abs`

Reply:
I think that [`abs` is overloaded for floating point types](https://www.cplusplus.com/reference/cmath/abs/), but I've changed it for clarity in https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1331/commits/5f587ea13c08992e745c97cb67cd9224e3f8d4a6.

Reply:
I believe `std::abs` is overloaded for floating-point types but the C function `::abs` only handles `int`

https://stackoverflow.com/a/33738702